public/app/core/components/TransformersUI/LabelsToFieldsTransformerEditor.tsx

```diff
@@ -8,14 +15,52 @@ export const LabelsAsFieldsTransformerEditor: React.FC<TransformerUIProps<Labels
   options,
   onChange,
 }) => {
-  return null;
+  let labelNames: Array<SelectableValue<string>> = [];
+  let uniqueLabels: Record<string, boolean> = {};
+
+  for (const frame of input) {
+    for (const field of frame.fields) {
+      if (!field.labels) {
+        continue;
+      }
+
+      for (const labelName of Object.keys(field.labels)) {
+        if (!uniqueLabels[labelName]) {
+          labelNames.push({ value: labelName, label: labelName });
+          uniqueLabels[labelName] = true;
+        }
+      }
+    }
+  }
+
+  const onValueLabelChange = (value: SelectableValue<string> | null) => {
+    onChange({ valueLabel: value?.value });
+  };
+
+  return (
+    <div className="gf-form-inline">
+      <div className="gf-form">
+        <div className="gf-form-label width-8">Value field name</div>
+        <Select
+          isClearable={true}
+          allowCustomValue={false}
+          placeholder="(Optional) Select label"
+          options={labelNames}
+          className="min-width-18 gf-form-spacing"
+          value={options?.valueLabel}
+          onChange={onValueLabelChange}
+          menuPlacement="bottom"
+        />
```

Comment:
Would be great to add some information about value field name behavior in the UI